src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_SIDE_LENGTH = 101;
+    private final Figure defaultFigure = new Circle(Color.WHITE, 10);
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    private enum Figures {
+        CIRCLE, RECTANGLE, SQUARE, RIGHT_TRIANGLE, ISOSCELES_TRAPEZOID
+    }
+
+    private final Figures[] figures = Figures.values();
+
+    public Figure getRandomFigure() {
+        Figure figure = defaultFigure;
+        switch (figures[random.nextInt(figures.length)]) {
+            case CIRCLE:
+                figure = new Circle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIDE_LENGTH));
+                break;
+            case RECTANGLE:
+                figure = new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIDE_LENGTH), random.nextInt(MAX_SIDE_LENGTH));
+                break;
+            case SQUARE:
+                new Square(colorSupplier.getRandomColor(), random.nextInt(MAX_SIDE_LENGTH));
```

Comment:
Looks like you forgot to assign `new Square(..) `to `figure`. Also, i think we can improve current solution:
I believe we do not need `figure` variable. In each case we can return new instance directly, and in the default clause - return `defaultFigure`.